docs/index.rst

```diff
@@ -18,6 +18,40 @@ Install from pypi::
 
     python -m pip install django-pg-migration-tools
 
+Background
+==========
+
+In the past, traditional applications would do schema changes during scheduled
+"maintenance windows." During this time, the application would stop all
+traffic, and developers could make changes to the database or servers without
+worrying about users.
+
+In the context of modern systems requiring 24/7 availability, this is no longer
+possible.
+
+Django's migration system usually picks the easiest way to make a schema
+change, but that isn't always safe. In fact, some operations Django does can
+cause problems, like locking tables for too long, which can cause outages.
+
+Modern applications also tend to make schema changes before deploying new code.
+For example, in a blue/green deployment, traffic is slowly moved from old
+servers (blue) to new servers (green).
+
+Once all traffic is migrated to the "green" servers, the "blue" servers usually
+remain ready in case the application needs to be rolled back.
```

Comment:
This examnple is missing information about when the schema changes are made.

Suggestion:

> For example, in a blue/green deployment, a schema change is made to the database, and then traffic is slowly moved from old servers (blue) to new servers (green).

Reply:
Good idea. Changed here: f5ace49c67facbaeb4971868cbcbc1f7b5423cac